linode/resource_linode_image.go

```diff
@@ -119,16 +157,29 @@ func resourceLinodeImageRead(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
-func resourceLinodeImageCreate(d *schema.ResourceData, meta interface{}) error {
+func resourceLinodeImageCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
create should wait for the image to have available status if uploaded. Otherwise, the user will have a non-empty plan after pulling in the status update